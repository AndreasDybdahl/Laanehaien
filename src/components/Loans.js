import React from 'react';
import PropTypes from 'prop-types';

export const Loans = ({loans, toggleInstallments}) => (
    <div>
        {loans.map((loan, index) => (
            <div key={index}>
                {JSON.stringify(loan)}
                <button onClick={() => {
                	toggleInstallments(loan.id)
                }}>
                    Slå {loan.paydownMonths === Number.MAX_VALUE ? "på" : "av"} nedbetaling
                </button>
            </div>
        ))}
    </div>
);

Loans.propTypes = {
    loans: PropTypes.array.isRequired,
    toggleInstallments: PropTypes.func
};