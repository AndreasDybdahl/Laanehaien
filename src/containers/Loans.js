import {connect} from 'react-redux';
import {Loans} from '../components/Loans'
import {getLoans} from '../modules/index';
import {toggleInstallments} from '../modules/loans';

const mapDispatchToProps = (dispatch) => {
    return {
        toggleInstallments: (loanId) => {
            dispatch(toggleInstallments(loanId))
        }
    }
};

export default connect(
    state => ({
        loans: getLoans(state)
    }),
    mapDispatchToProps
)(Loans);