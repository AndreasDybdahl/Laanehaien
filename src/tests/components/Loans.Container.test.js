import React from 'react';
import renderer from 'react-test-renderer';
import { 
    BrowserRouter as Router,
} from 'react-router-dom';
import {Provider} from 'react-redux';
import Loans from '../../containers/Loans';
import {createStore, applyMiddleware, compose} from 'redux';

const store = createStore(
    () => { return {loans: []}},
);

it('should render the loans container', () => {
    const tree = renderer.create(
         <Provider store={store}>
        <Router>
            <Loans />
        </Router>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
})