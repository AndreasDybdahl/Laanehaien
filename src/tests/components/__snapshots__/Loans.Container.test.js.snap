exports[`test should render the loans container 1`] = `<div />`;
